python/ray/data/grouped_dataset.py

```diff
@@ -144,6 +146,111 @@ def _aggregate_on(
         )
         return self.aggregate(*aggs)
 
+    def map_groups(
+        self,
+        fn: Union[CallableClass, Callable[[BatchType], BatchType]],
+        *,
+        compute: Optional[str] = None,
+        batch_format: str = "native",
+        **ray_remote_args
+    ) -> "Dataset[Any]":
+        """Apply the given function to each group of records of this dataset.
+
+        While the map_groups() is very flexibile, note that it comes with downsides:
+            - It may be slower than using more specific methods such as min(), max().
+            - It requires the size of each group fit on a single node.
+        So when specific methods are applicable, try to use them before reaching
+        map_groups().
+
+        This is a blocking operation.
+
+        Examples:
+            >>> # Compute an arbitrary function on each group.
+            >>> ray.data.range(100).groupby(lambda x: x % 3).map_groups(
+                    lambda x: [min(x) / max(x)])
```

Comment:
Can you add examples of return multiple records per group? Add comments on how you can return one item per group, or multiple items.

Reply:
Done. LMK if it looks good.